Type error in @trpc/server with TS 4.6

Hi @KATT,  
Thank you for this great library.  

I am reporting that the server types need updates after the latest typescript release (4.6):  

```bash
node_modules/@trpc/server/dist/declarations/src/TRPCError.d.ts:7:14 - error TS2416: Property 'cause' in type 'TRPCError' is not assignable to the same property in base type 'Error'.
  Type 'unknown' is not assignable to type 'Error | undefined'.

7     readonly cause?: unknown;
               ~~~~~


Found 1 error in node_modules/@trpc/server/dist/declarations/src/TRPCError.d.ts:7
```

![image](https://user-images.githubusercontent.com/6702424/156262690-0552b0f1-4b45-4172-b39c-6d5204b66219.png)

Best regards

--- a/packages/server/src/TRPCError.ts
+++ b/packages/server/src/TRPCError.ts
@@ -6,7 +6,7 @@ export class TRPCError extends Error {
    * @deprecated use `cause`
    */
   public readonly originalError?: unknown;
-  public readonly cause?: unknown;
+  public readonly cause?;
   public readonly code;
 
   constructor(opts: {
@@ -16,6 +16,8 @@ export class TRPCError extends Error {
      * @deprecated use `cause`
      */
     originalError?: unknown;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore This will be `Error` in next major version
     cause?: unknown;
   }) {
     const cause = opts.cause ?? opts.originalError;

